app/models/package_manager/maven/maven_central.rb

```diff
@@ -20,4 +20,20 @@ def self.one_version(raw_project, version_string)
   def self.missing_version_remover
     PackageManager::Base::MissingVersionRemover
   end
+
+  # maven-metadata.xml for Maven Central does not appear to be guaranteed to contain all relevant versions for a package
+  # So instead, if needed, we will retrieve the versions from the raw HTML index page
+  def self.versions(raw_project, name)
+    if raw_project && raw_project[:versions]
+      raw_project[:versions]
+    else
+      retrieve_versions(versions_from_html(name), name)
+    end
+  end
+
+  def self.versions_from_html(name)
+    get_html(MavenUrl.from_name(name, repository_base, NAME_DELIMITER).base).css("a").filter_map do |a|
+      a.text.chomp("/") if a.text.ends_with?("/") && a.text != "../"
```

Comment:
Do we know how reliable the folder structure maps to actual versions/artifacts (i.e. are all folders versions and all versions folders?)?

Reply:
Technically no, we do not reliably know this. But this should not be an issue because in `retrieve_versions` we handle cases where we do not find a version `.pom` at the correct address, so any invalid matches should be handled. https://github.com/librariesio/libraries.io/blob/main/app/models/package_manager/maven.rb#L195-L198

Reply:
Can we log something if what is expected to be a real version based off the string is not actually a real version so we can improve the regex matching in the future?

Reply:
I have added a log here if a version is not found, in case the directory structure includes more directories we don't expect.